test/user_test.rb

```diff
@@ -0,0 +1,46 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require_relative '../lib/joke'
+require_relative '../lib/user'
+
+class JokeTest < Minitest::Test
+
+  def test_it_exists
+    sal = User.new("Sal")
+    assert_instance_of User, sal 
+  end 
+  
+  def test_it_has_attributes
+    sal = User.new("Sal")
+    assert_equal "Sal", sal.name
+    assert_equal [], sal.jokes
+     assert_equal [], sal.jokes_id
+  end
+
+
+  def test_it_can_learn_new_joke
+    sal = User.new("Sal")
+    joke_1 = Joke.new(1, "Why did the strawberry cross the road?", "Because his mother was in a jam.")
+    joke_2 = Joke.new(2, "How do you keep a lion from charging?", "Take away its credit cards.")    
+    assert_equal [], sal.jokes
+    sal.learn(joke_1)
+    assert_equal [joke_1], sal.jokes
+    sal.learn(joke_2)
+    assert_equal [joke_1, joke_2], sal.jokes
```

Comment:
since you also have your `learn` method updating the `@jokes_id` attribute, i would test that here as well, especially since your `repeated_jokes?` method in the OpenMic class relies on that functionality working correctly.